### ISSUE TITLE
linter: extglob `@(a|b|c)` patterns in `overrides[].files` are silently ignored

### ISSUE THREAD

**Issue by B4nan** [oxc-project/oxc#21525]
### What version of Oxlint are you using?

```
1.60.0
```

### What command did you run?

```
oxlint test.stories.tsx
```

### What does your `.oxlintrc.json` config file look like?

Two configs that differ only in the `files` glob:

**A — brace expansion `{ts,tsx,...}`** (works):

```jsonc
{
  "overrides": [
    {
      "files": ["**/*.stories.{tsx,ts}"],
      "rules": { "no-console": "error" }
    }
  ]
}
```

**B — extglob `@(ts|tsx|...)`** (broken):

```jsonc
{
  "overrides": [
    {
      "files": ["**/*.stories.@(ts|tsx|js|jsx|mjs|cjs)"],
      "rules": { "no-console": "error" }
    }
  ]
}
```

### What happened?

Test file `test.stories.tsx`:

```ts
console.log('hi');
```

**Config A** (brace expansion):

```
x eslint(no-console): Unexpected console statement.
   ,-[test.stories.tsx:1:1]
 1 | console.log('hi');
   : ^^^^^^^^^^^

Found 0 warnings and 1 error.
```

**Config B** (extglob):

```
Found 0 warnings and 0 errors.
```

The override silently has no effect. No warning that the pattern didn't compile, no diagnostic that no files matched the override — the rule is just dropped.

The two glob patterns are semantically equivalent (same set of file extensions) and both are valid ESLint `files` patterns — ESLint accepts and applies both forms identically.

### Why this matters in practice

The `**/*.stories.@(ts|tsx|js|jsx|mjs|cjs)` form is the **default pattern** in `eslint-plugin-storybook`'s flat config (`csf_default2.files` in `eslint-plugin-storybook@10.3.1`'s `dist/index.js`). Anyone copying that config into oxlint will silently lose all storybook rule enforcement and have no way to notice without writing test cases.

The same issue applies to all `@(...)` extglob patterns regardless of plugin — this is a general `overrides.files` matcher bug, not a plugin-specific one. I happened to hit it via storybook but the repro above uses the built-in `no-console` rule to keep it minimal.

### Expected behavior

Either:
1. `@(a|b|c)` extglob patterns in `overrides[].files` should match the same files as `{a,b,c}` brace expansion (matches ESLint behavior), or
2. If the extglob syntax is intentionally not supported, oxlint should emit an error or warning when it encounters one, instead of silently producing an unmatchable pattern.

### Environment

|                |                             |
| -------------- | --------------------------- |
| oxlint         | 1.60.0                      |
| Node.js        | v22.13.0                    |
| OS             | macOS Darwin 25.4.0 (arm64) |

### Minimal reproduction

```bash
mkdir oxlint-extglob-bug && cd oxlint-extglob-bug
npm init -y >/dev/null && npm i -D oxlint@1.60.0 >/dev/null
echo "console.log('hi');" > test.stories.tsx

cat > .oxlintrc.json <<'JSON'
{ "overrides": [{ "files": ["**/*.stories.@(ts|tsx)"], "rules": { "no-console": "error" } }] }
JSON
npx oxlint test.stories.tsx
# -> Found 0 warnings and 0 errors  (BUG: rule not applied)

cat > .oxlintrc.json <<'JSON'
{ "overrides": [{ "files": ["**/*.stories.{ts,tsx}"], "rules": { "no-console": "error" } }] }
JSON
npx oxlint test.stories.tsx
# -> Found 0 warnings and 1 error  (works)
```

**Comment by leaysgur** [comment]
NOTE:

- The point is should we support so called `extglob`?
- Oxlint, and also Oxfmt use `fast-glob` to match globs for `overrides`
  - https://github.com/oxc-project/fast-glob
- `fast-glob` does not support `extglob` for now
  - And issue related to `extglob` syntax seemed to be closed in the past
  - Is it possible to detect without perf hit...?

UPDATE: Invalid patterns are now validated, extglob is not addressed yet.
https://github.com/oxc-project/fast-glob/pull/153

**Comment by camc314** [comment]
@leaysgur I don't think we should fix this by changing the globb-ing library, but it's wierd that we don't flag a warning/error on this case - lets discuss tomorow?

**Comment by mymx2** [comment]
Hit the same thing from a different angle on 1.76.0 (Windows): the `?()` optional-group form `**/*.test.?([cm])[jt]s?(x)` also silently matches nothing. That's the shape `antfu/eslint-config` uses for its test globs, so it's another common port-in footgun beyond `@()`.

One extra observation that made this especially confusing to debug: when the override silently doesn't apply, a rule that's on by category default (e.g. `vitest/no-focused-tests` at warn) still fires — at its default severity. So it doesn't even look like the override is broken; it looks like your `error` was downgraded to `warn`. Only rules that are off by default (like `consistent-test-it`) reveal the truth by not running at all. That mix makes a silent no-match really hard to spot, so +1 for the warn-on-unsupported-pattern direction camc314 suggested.

**Comment by leaysgur** [comment]
Thanks for the report.
I took some time to look into this.

- Properly supporting `extglob` in `fast-glob` is not easy:
  - `@(a|b)` and `?(a|b)` could be mapped onto the existing brace-expansion machinery
  - but `*(...)`, `+(...)` and `!(...)` cannot
- Supporting them would break `fast-glob`'s O(1)-space backtracking design
  - it would kill the "fast" part and essentially require a different, regex-engine-like implementation
- I think we should not support it in `fast-glob`

If we want to warn these, the catches are:

- every `extglob` pattern is also a syntactically valid plain glob (`@`, `(`, `|`, ) are all treated as literals)
- and in particular `?(` / `*(` are indistinguishable from a legitimate "wildcard + literal `(`" pattern
  - e.g. `file?(1).txt`

That said, this is a niche enough case that a heuristic, flag any "extglob-looking" prefix followed by a matching `)` with a warning seems harmless in practice. False positives can be silenced by escaping the literal paren as `\(`.

Also, since there is currently no mechanism to report config issues as warnings rather than errors, we need to address that as well.

Alternatively, we could just document this limitation and leave it at that?
